File: support_tools/swigwin-3.0.12/Examples/test-suite/python/funcptr_cpp_runme.py
from funcptr_cpp import *

if call1(ADD_BY_VALUE, 10, 11) != 21:
    raise RuntimeError
if call2(ADD_BY_POINTER, 12, 13) != 25:
    raise RuntimeError
if call3(ADD_BY_REFERENCE, 14, 15) != 29:
    raise RuntimeError
if call1(ADD_BY_VALUE_C, 2, 3) != 5:
    raise RuntimeError

File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/funcptr_cpp_runme.lua
require("import")	-- the import fn
import("funcptr_cpp")	-- import lib into global
fc=funcptr_cpp --alias

-- catching undefined variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

assert(fc.addByValue(5,10) == 15)
-- These two won't work. Lua will successfully store the answer as userdata, but there is
-- no way of accessing the insides of userdata.
-- assert(fc.addByPointer(7, 9) == 16)
-- assert(fc.addByReference(8, 9) == 17)

assert(fc.call1(fc.ADD_BY_VALUE, 5, 10) == 15)
assert(fc.call2(fc.ADD_BY_POINTER, 7, 9) == 16)
assert(fc.call3(fc.ADD_BY_REFERENCE, 8, 9) == 17)
assert(fc.call1(fc.ADD_BY_VALUE_C, 2, 3) == 5)
